### PR TITLE
Creating training stages of pipeline and notebook to run

### LINKED ISSUE #9: Automatic reproduction of experiment
We should change the way we defined either the pipeline's scripts or the Colab notebook to allow the reproduction of our experiment without any human intervention: at the moment to re-launch the experiment we need to
1. pre-process the data;
2. upload the data to Google Colab;
3. rearrange the data in a way that is useful to work with in Colab;
4. setting the `DATA_PATH` variable on Colab
5. launching the iPyNB.

Although at this stage the build is actually reproducible but with some human intervention, we should automatise it as much as possible: we should be able to relaunch everything by simply changing the parameters and pressing "launch" (or invoking a command).

### PR DESCRIPTION
**Pull request by montanarograziano**
In this PR the following stages of the pipeline are implemented:
- evaluate.py
- train.py
- model.py (only as a library purpose)
Also, the run_all.ipynb is added to shift the workflow on Google Colab.
This closes #9 .

### CODE REVIEW

**Comment by espositoandrea** [comment]
I've reformatted some files using the PEP8 style guide so everything should now be more readable.

File: src/train.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    args.data,
+    validation_split=params['validation_split'],
+    subset="validation",
+    seed=params['seed'],
+    shuffle=True,
+    image_size=(128, 128),
+    color_mode="grayscale",
+    label_mode="binary",
+    batch_size=params['batch_size']
+)
```
Comment (espositoandrea) [inline src/train.py]: The final training phase should not have a validation set... I'll fix it since I'm already looking at this code 😁

File: src/train.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+checkpoint_cb = keras.callbacks.ModelCheckpoint(
+    filename, save_best_only=True
+)
+early_stopping_cb = keras.callbacks.EarlyStopping(
+    monitor="val_accuracy", patience=params["patience"])
+
+# Train the model, doing validation at the end of each epoch
+history = model.fit(
+    train_ds,
+    validation_data=val_ds,
```
Comment (espositoandrea) [inline src/train.py]: As above: the final training phase should not have a validation set. I'll fix it since I'm already looking at this code 😉

**Comment by espositoandrea** [comment]
I've fixed the thing I highlighted in the comments, so I'm marking this PR as ready to merge. Actually, I'm merging it straight away since it blocks #11.

**Review by espositoandrea — APPROVED** [review]
